### PR TITLE
fix: app.go, error coding, scripts

### PR DESCRIPTION
**Pull request by charleenfei**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,26 +1,20 @@
...
+###############################################################################
+###                     Install & Clean                                     ###
+###############################################################################
 
 all: install
 
 install: go.sum
-		go install $(BUILD_FLAGS) ./cmd/pooltoyd
+	@echo "--> installing pooltoy"
+	@go install -v -mod=readonly ./...
```
Comment (marbar3778) [inline Makefile:13]:
```suggestion
	@go install -v -mod=readonly ./cmd/pooltoyd
```

Should we keep the build flags as well?

File: app/types.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/server/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// App implements the common methods for a Cosmos SDK-based application
+// specific blockchain.
+type CosmosApp interface {
```
Comment (marbar3778) [inline app/types.go:13]: can this be imported from sdk?

File: app/app.go [hunk 3]
```diff
@@ -63,253 +80,256 @@ import (
...
+		gov.NewAppModuleBasic(getGovProposalHandlers()...),
 		params.AppModuleBasic{},
-		upgrade.AppModuleBasic{},
+		crisis.AppModuleBasic{},
 		slashing.AppModuleBasic{},
+		ibc.AppModuleBasic{},
+		upgrade.AppModuleBasic{},
+		evidence.AppModuleBasic{},
+		transfer.AppModuleBasic{},
+		vesting.AppModuleBasic{},
```
Comment (marbar3778) [inline app/app.go:142]: we can get rid of this

File: Makefile (hunk 1 above)
Comment (charleenfei) [reply Makefile:13]: yea true, will put those back in

File: app/app.go (hunk 3 above)
Comment (charleenfei) [reply app/app.go:140]: i need evidence and transfer for ibc integration

File: app/types.go (hunk 2 above)
Comment (charleenfei) [reply app/types.go]: yea, good point

File: app/app.go (hunk 3 above)
Comment (marbar3778) [reply app/app.go:140]: oh, I meant vesting. Unless we have some vesting accounts on pooltoy? Its more of a nit to not instantiate modules we dont use.

File: app/app.go [hunk 4]
```diff
@@ -63,254 +82,251 @@ import (
...
+		mint.AppModuleBasic{},
 		distr.AppModuleBasic{},
-		gov.NewAppModuleBasic(
-			paramsclient.ProposalHandler,
-			distrclient.ProposalHandler,
-			upgradeclient.ProposalHandler,
-		),
+		gov.NewAppModuleBasic(getGovProposalHandlers()...),
 		params.AppModuleBasic{},
-		upgrade.AppModuleBasic{},
```
Comment (marbar3778) [inline app/app.go:103]: do we want to keep upgrade?

File: app/app.go [hunk 5]
```diff
@@ -3,32 +3,46 @@ package app
...
 	"github.com/cosmos/cosmos-sdk/types/module"
 	"github.com/cosmos/cosmos-sdk/version"
 	"github.com/cosmos/cosmos-sdk/x/auth"
 	"github.com/cosmos/cosmos-sdk/x/auth/ante"
 	authrest "github.com/cosmos/cosmos-sdk/x/auth/client/rest"
 	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authsims "github.com/cosmos/cosmos-sdk/x/auth/simulation"
 	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting"
```
Comment (marbar3778) [inline app/app.go:36]: vesting is being used somewhere, should remove.

**Review by marbar3778 — APPROVED** [review]

File: app/app.go (hunk 4 above)
Comment (charleenfei) [reply app/app.go:103]: i thought i removed it 🙈

**Pull request merged**